MAVProxy/mavproxy.py

```diff
@@ -561,11 +561,12 @@ def process_mavlink(slave):
         return
     if msgs is None:
         return
-    if mpstate.settings.mavfwd and not mpstate.status.setup_mode:
-        for m in msgs:
-            if mpstate.status.watch is not None:
-                if fnmatch.fnmatch(m.get_type().upper(), mpstate.status.watch.upper()):
+    if mpstate.settings.mavfwd and not mpstate.status.setup_mode and mpstate.status.watch:
+        for msg_type in mpstate.status.watch:
+            for m in msgs:
```

Comment:
I think this still isn't OK. The for loops should be switched. What we want is for all the messages we have received (`msgs`), check if it matches at most one watch regex that has been set - what it does now is for every regex that has been set check if there is at most one received message that matches.

Reply:
This way will avoid unnecessary loop interactions.
I did a python script if you want to try :)